addons/EditableWindow/src/presenter.js

```diff
@@ -0,0 +1,455 @@
+            audioFile: model['audio'],
+            videoFile: model['video'],
+        }
+    };
+
+    presenter.generateValidationError = function (message) {
+        return {
+            isValid: false,
+            errorMessage: message
+        }
+    };
+
+    presenter.fetchIframeContent = function (callback) {
+        var view = presenter.configuration.view;
+        var isIframeLoaded = presenter.configuration.isIframeLoaded;
+
+        if (isIframeLoaded) {
+            var content = $(view).find(".content-iframe").contents().find("body").html();
+            if (content == null || content == "") {
+                var timeout = setTimeout(function () {
+                    presenter.fetchIframeContent(callback);
+                }, 1000);
+                presenter.configuration.timeouts.push(timeout);
+            } else {
+                callback(content);
+            }
+        } else {
+            var timeout = setTimeout(function (callback) {
+                presenter.fetchIframeContent(callback)
+            }, 1000);
+
+            presenter.configuration.timeouts.push(timeout);
+        }
+    };
+
+    presenter.fillActiveTinyMce = function (content) {
+        var isTinyMceLoaded = presenter.configuration.isTinyMceLoaded;
+
+        if (isTinyMceLoaded) {
+            var timeout = setTimeout(function () {
+                presenter.fillTinyMce(content);
+            }, 1000);
+
+            presenter.configuration.timeouts.push(timeout);
+        } else {
+            var timeout = setTimeout(function () {
+                presenter.fillActiveTinyMce(content);
+            }, 1000);
+
+            presenter.configuration.timeouts.push(timeout);
+        }
+    };
+
+    presenter.fillTinyMce = function (content) {
+        var fileList = presenter.configuration.model.fileList;
+        var documentContent = new DOMParser().parseFromString(content, 'text/html');
+        var textareaId = presenter.configuration.textareaId;
+
+        for (var i = 0; i < fileList.length; i++) {
+            var entity = fileList[i];
+            var node = documentContent.getElementById(entity.id);
+            if (node != null && node !== undefined) {
+                node.src = entity.file;
+            }
+        }
+
+        var parsedContent = documentContent.getElementsByTagName("body")[0].innerHTML;
+        tinymce.get(textareaId).getBody().innerHTML = parsedContent;
+        presenter.configuration.isTinyMceFilled = true;
+    };
+
+    presenter.removeIframe = function () {
+        $(presenter.configuration.view).find(".content-iframe").remove();
+    };
+
+    presenter.centerPosition = function () {
+        var $view = $(presenter.configuration.view);
+        var width = $view.find(".addon-editable-window-wrapper").width();
+        var icPageWidth = $(".ic_page").width();
+
+        var scrollY;
+        try {
+            scrollY = window.parent.scrollY !== undefined ? window.parent.scrollY : window.scrollY;
```

Comment:
Tu można by wykorzystać: presenter.configuration.playerController.iframeScroll();
Ta funkcja powinna zwracać to samo co ten fragment i sprawdza również CORSy.
Scroll dla playera moze być ustawiony z zewnątrz przez odpowiedni message - ta funkcja z playerControllera będzie wtedy przesyłała tę wartość ustawioną z zewnątrz.